src/lib/doctrine-dbal-bulk/src/Flow/Doctrine/Bulk/Dialect/MySQLDialect.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\Doctrine\Bulk\Dialect;
+
+use Flow\Doctrine\Bulk\BulkData;
+use Flow\Doctrine\Bulk\Columns;
+use Flow\Doctrine\Bulk\Exception\RuntimeException;
+use Flow\Doctrine\Bulk\TableDefinition;
+
+final class MySQLDialect implements Dialect
+{
+    /**
+     * @param TableDefinition $table
+     * @param BulkData $bulkData
+     * @param array{
+     *  skip_conflicts?: boolean,
+     * } $insertOptions $insertOptions
+     *
+     * @return string
+     */
+    public function prepareInsert(TableDefinition $table, BulkData $bulkData, array $insertOptions = []) : string
+    {
+        if (\array_key_exists('skip_conflicts', $insertOptions) && $insertOptions['skip_conflicts'] === true) {
+            return \sprintf(
+                'INSERT INTO %s (%s) VALUES %s ON DUPLICATE KEY UPDATE %4$s=%4$s',
+                $table->name(),
+                $bulkData->columns()->concat(','),
+                $bulkData->toSqlPlaceholders(),
+                \current($bulkData->columns()->all())
+            );
+        }
+
+        return \sprintf(
+            'INSERT INTO %s (%s) VALUES %s',
+            $table->name(),
+            $bulkData->columns()->concat(','),
+            $bulkData->toSqlPlaceholders()
+        );
+    }
+
+    /**
+     * @param TableDefinition $table
+     * @param BulkData $bulkData
+     * @param array{
+     *  update_columns?: array<string>
+     * } $updateOptions $updateOptions
+     *
+     * @throws RuntimeException
+     *
+     * @return string
+     */
+    public function prepareUpdate(TableDefinition $table, BulkData $bulkData, array $updateOptions = []) : string
+    {
+        return \sprintf(
+            'INSERT INTO %s (%s) 
+            VALUES %s 
+            ON DUPLICATE KEY UPDATE %s',
```

Comment:
So this should actually go to the prepareInsert as you are doing an INSERT here.
This is how I would handle upserts in mysql:

* INSERT IGNORE - preapreInsert
* INSERT ON DUPLICATE KEY UPDATE - prepare insert
* REPLACE - prepareUpdate

Reply:
OK, I've fixed it like that.